check-module-version/action.yml

```diff
@@ -0,0 +1,51 @@
+name: 'Check module version'
+description: >
+  Сhecks the equality of the version from Lua 
+  module's code and the last repository tag
+
+inputs:
+  module-name:
+    description: 'Lua module name as when using require() in tarantool'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Clone the module
+      uses: actions/checkout@v3
+
+    - name: Setup tarantool
+      uses: tarantool/setup-tarantool@v2
+      with:
+        tarantool-version: '2.10'
+
+    - name: Make rock
+      run: tarantoolctl rocks make
+      shell: bash
+
+    - name: Code version
+      run: |
+        MODULE_VERSION=$(tarantool -e "print(require('${{ inputs.module-name }}')._VERSION)")
+        echo Module name is ${{ inputs.module-name }}
+        echo Detected version from code is $MODULE_VERSION
+        if [ "$MODULE_VERSION" = "nil" ]; then
+          echo "Check the version value implementation in the Lua module's code"
+          exit 1
+        fi
+        echo "MODULE_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Package version
+      # We need this step to run only on push with tag.
+      if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
```

Comment:
Honestly, I don't like that we use such a logic in the action. Looks like this action should be run when this condition is true. Shouldn't it be like this in your workflow file:
```
jobs:
  version-check:
    if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
    runs-on: ubuntu-20.04
    steps:
      - name: Run action
        uses: tarantool/actions/check-module-version@master
        with:
          # Need to use the actual name instead <foobar>.
          module-name: 'foobar'
```
?

Reply:
Hmm, if we use this `if` condition for the whole `version-check`, then this job will only work on the tag.

As far as I understand, it would be good to get a check that hardcoded versioning is generally implemented in the module (step `Code version`) for regular (not tag) commits

@DifferentialOrange Do I understand the idea correctly?

Reply:
Yes, unfortunately, we cannot use simpler logic for our general purpose solution. The reasons are as follows:
- steps from this action are best when they are a separate job, since they mess with `.rocks` contents,
- most of our modules do not separate packaging from publishing: they have a single workflow "package for each push on master, but publish on workflow" (especially for enterprise repos, when you can't use GitHub Actions artifacts to transfer package between separate packaging and publishing step).

Since there is no on-board way to make "run on push and tag" job depend on "run only on tag" job, we decided that the least painful way to satisfy both of these conditions is to have an action which can run both for tags and non-tags.

Reply:
On the other hand, we cannot build an action which will work without tag conditions, since release commit would always fail such step (and we don't have any rules for release commits to workaround it anyway).

Reply:
> it would be good to get a check that hardcoded versioning is generally implemented in the module (step Code version) for regular (not tag) commits

It is good, but it wasn't the main motivation.

Reply:
Updated